c7n/resources/iam.py

```diff
@@ -536,6 +536,43 @@ def process(self, resources, event=None):
             len(results), len(resources))
         return results
 
+
+@Policy.action_registry.register('delete')
+class PolicyDelete(BaseAction):
+    """Delete an IAM Policy.
+
+    For example, if you want to automatically delete all unused IAM policies.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: iam-delete-unused-policies
+          resource: iam-policy
+          filters:
+            - type: unused
+            - type: value
+                key: "Arn"
+                op: regex
+                value: "(arn:aws:iam::[0-9])"
+                # regular expression to filter out AWS managed policies.
+                # they are in the format arn:aws:iam::aws
+                # user policies are in the format arn:aws:iam::${account_number}
+          actions:
+            - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeletePolicy',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        for r in resources:
+            self.log.debug('Deleting policy %s' % r['PolicyName'])
```

Comment:
lgtm, but should we be filtering out managed policies in the action as a safety guard?

Reply:
That's a good point; you can't delete managed policies. I'll update the PR to reflect this.